lib/utils/aws/aws.go

```diff
@@ -208,3 +209,41 @@ func filterHeaders(r *http.Request, headers []string) {
 	}
 	r.Header = out
 }
+
+// FilterAWSRoles returns role ARNs from the provided list that belong to the
+// specified AWS account ID.
+//
+// If AWS account ID is empty, all roles are returned.
+func FilterAWSRoles(arns []string, accountID string) (result []AWSRole) {
+	for _, roleARN := range arns {
+		parsed, err := arn.Parse(roleARN)
+		if err != nil || (accountID != "" && parsed.AccountID != accountID) {
```

Comment:
Should we return `err` if it's not nil? Aren't we silently ignoring parsing errors otherwise?

Reply:
I didn't write these and don't use them here, just moved them out of `lib/srv/app` so I could use a couple things and put all the aws helpers in one place. But I can probably take a look

Reply:
Let's leave those as a nice-to-have for a separate PR, then. Feel free to close the comments.